### DISCUSSION TITLE
Hydration completed but contains mismatches, on component loading with custom path.

### DISCUSSION THREAD

**Discussion by mrFANRA** [nuxt/nuxt#21039] [Questions]
Hi!

I have new Nuxt3 empty project. Got error, when trying import component(not auto-loaded) from custom path.  
It is look like component not rendered on server, view source showing content without component.

But when I use <script setup> style all is working fine. On preview mode all is work. When I use component folder with autoload is working fine too.

Why it is happened? Help me please...

/pages/index.vue
```
<template>
  <div><Demo/></div>
</template>

<script>
import Demo from "~/my-components/demo.vue";
export default {
  name: "index",
  components:{Demo},
}
</script>
```

/my-components.vue
```
<template>
  <div>
    im demo
  </div>
</template>

<script>
export default {
  name: "Demo",
}
</script>
```

**package.json**
```
{
  "name": "nuxt-app",
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "overrides": {
    "vue": "latest"
  },
  "devDependencies": {
    "nuxt": "^3.5.1"
  }
}

```

**Vite console error:**
```
[Vue warn]: resolveComponent can only be used in render() or setup().                                                                                                                                                                
[Vue warn]: Component <Anonymous> is missing template or render function.
```

**Browser error**
```
[Vue warn]: Hydration node mismatch:
- Client vnode: div 
- Server rendered DOM: 
<!--  -->
 <empty string> 
  at <Demo> 
  at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > > 
  at <RouteProvider key="/" routeProps= 
Object { Component: {…}, route: {…} }
 pageKey="/"  ... > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage> 
  at <NuxtLayout> 
  at <App key=3 > 
  at <NuxtRoot> [deps:97:13](http://localhost:3000/_nuxt/node_modules/.cache/vite/client/deps/)
Hydration completed but contains mismatches. [deps:4572:15](http://localhost:3000/_nuxt/node_modules/.cache/vite/client/deps/)
```

**Screens**
![Screenshot_3](https://github.com/nuxt/nuxt/assets/20259213/d1277708-7288-4e79-8235-15594b9215e0)

![Screenshot_2](https://github.com/nuxt/nuxt/assets/20259213/ac3440f1-3789-413f-92fa-6c2ad823a74e)

![Screenshot_1](https://github.com/nuxt/nuxt/assets/20259213/c81592a6-7422-4191-836e-5ea0de753bc2)

**Comment by huang-julien** [comment]
I can't reproduce it, can you provide a minimal reproduction please ? you can use stackblitz 
https://stackblitz.com/edit/nuxt-starter-vrzcif?file=app.vue,.gitignore

**Reply by mrFANRA** [reply]
Nothing understand. It work in reproduction...

https://stackblitz.com/edit/nuxt-starter-bcx2rv?file=pages%2Findex.vue

**Reply by mrFANRA** [reply]
It is strange that script setup style is work! =(

```
<template>
  <div><Demo/></div>
</template>

<script setup>
import Demo from "../my-components/demo.vue";
</script>
```
